netket/hilbert/abstract_hilbert.py

```diff
@@ -59,124 +59,25 @@ def __init__(self):
-    ) -> np.ndarray:
-        r"""Returns the quantum numbers corresponding to the n-th basis state
-        for input n. n is an array of integer indices such that
-        :code:py:`numbers[k]=Index(states[k])`.
-        Throws an exception iff the space is not indexable.
-
-        Args:
-            numbers (numpy.array): Batch of input numbers to be converted into arrays of
-                quantum numbers.
-            out: Optional Array of quantum numbers corresponding to numbers.
-        """
-        if out is None:
-            out = np.empty((np.atleast_1d(numbers).shape[0], self.size))
-
-        if np.any(numbers >= self.n_states):
-            raise ValueError("numbers outside the range of allowed states")
-
-        if np.isscalar(numbers):
-            return self._numbers_to_states(np.atleast_1d(numbers), out=out)[0, :]
-        else:
-            return self._numbers_to_states(numbers, out=out)
-
-    def states_to_numbers(
-        self, states: np.ndarray, out: Optional[np.ndarray] = None
-    ) -> Union[int, np.ndarray]:
-        r"""Returns the basis state number corresponding to given quantum states.
-        The states are given in a batch, such that states[k] has shape (hilbert.size).
-        Throws an exception iff the space is not indexable.
-
-        Args:
-            states: Batch of states to be converted into the corresponding integers.
-            out: Array of integers such that out[k]=Index(states[k]).
-                 If None, memory is allocated.
-
-        Returns:
-            numpy.darray: Array of integers corresponding to out.
-        """
-        if states.shape[-1] != self.size:
-            raise ValueError(
-                f"Size of this state ({states.shape[-1]}) not"
-                f"corresponding to this hilbert space {self.size}"
-            )
-
-        states_r = np.asarray(np.reshape(states, (-1, states.shape[-1])))
-
-        if out is None:
-            out = np.empty(states_r.shape[:-1], dtype=np.int64)
-
-        out = self._states_to_numbers(states_r, out=out.reshape(-1))
-
-        if states.ndim == 1:
-            return out[0]
-        else:
-            return out.reshape(states.shape[:-1])
+        return False  # pragma: no cover
 
     @property
     def n_states(self) -> int:
         r"""The total dimension of the many-body Hilbert space.
         Throws an exception iff the space is not indexable."""
-        raise NotImplementedError()
-
-    def states(self) -> Iterator[np.ndarray]:
-        r"""Returns an iterator over all valid configurations of the Hilbert space.
-        Throws an exception iff the space is not indexable.
-        Iterating over all states with this method is typically inefficient,
-        and ```all_states``` should be prefered.
-
-        """
-        for i in range(self.n_states):
-            yield self.numbers_to_states(i).reshape(-1)
+        raise ValueError("The Hilbert space is infinite.")
```

Comment:
Nitpick: I'd raise a `NotImplementedError` (subclasses of `AbstractHilbert` can be both be finite and infinite and it seems strange to have the superclass decide on one and override it in a subclass).